topostats/tracing/dnatracing.py

```diff
@@ -157,65 +159,41 @@ def _crop_array(array: np.ndarray, bounding_box: Tuple) -> np.ndarray:
-
-        Uses my own skeletonisation function from tracingfuncs module. I will
-        eventually get round to editing this function to try to reduce the branching
-        and to try to better trace from looped molecules"""
-
-        for grain_num in sorted(self.grains.keys()):
-            # What purpose does binary dilation serve here? Name suggests some sort of smoothing around the edges and
-            # the resulting array is used as a mask during the skeletonising process.
-            smoothed_grain = ndimage.binary_dilation(self.grains[grain_num], iterations=1).astype(
-                self.grains[grain_num].dtype
-            )
-
-            sigma = 0.01 / (self.pixel_size * 1e9)
-            very_smoothed_grain = ndimage.gaussian_filter(smoothed_grain, sigma)
-
+        """Puts skeletons into dictionary"""
+        for grain_num, skeleton in self.skeleton_dict.items():
             try:
-                dna_skeleton = getSkeleton(
-                    self.gauss_image, smoothed_grain, self.number_of_columns, self.number_of_rows, self.pixel_size
-                )
-                self.disordered_trace[grain_num] = dna_skeleton.output_skeleton
+                self.disordered_traces[grain_num] = np.argwhere(skeleton == 1)
             except IndexError:
-                # Some gwyddion grains touch image border causing IndexError
-                # These grains are deleted
+                # Grains touching border (IndexError) are deleted
                 self.grains.pop(grain_num)
-            # skel = morphology.skeletonize(self.grains[grain_num])
-            # self.skeletons[grain_num] = np.argwhere(skel == 1)
 
     def purge_obvious_crap(self):
-
-        for dna_num in sorted(self.disordered_trace.keys()):
-
-            if len(self.disordered_trace[dna_num]) < 10:
-                self.disordered_trace.pop(dna_num, None)
-
-    def linear_or_circular(self, traces):
-
-        """Determines whether each molecule is circular or linear based on the local environment of each pixel from the trace
-
-        This function is sensitive to branches from the skeleton so might need to implement a function to remove them"""
+        """Checks skeleton is <10 pixels, and removes otherwise."""
+        for dna_num in sorted(self.disordered_traces.keys()):
+            if len(self.disordered_traces[dna_num]) < 10:
+                self.disordered_traces.pop(dna_num, None)
+
+    def linear_or_circular(self, traces: dict):
+        """Determines whether each molecule is circular or linear based on the local environment of each pixel from the trace.
+        This function is sensitive to branches from the skeleton so might need to implement a function to remove them
+        
+        Parameters
+        ----------
+        traces: dict
+            A dictionary of the molecule_number and points within the skeleton.
+        """
 
         self.num_circular = 0
         self.num_linear = 0
 
         for dna_num in sorted(traces.keys()):
```

Comment:
As a general comment I'm not a fan of looping over dictionary/lists within a method.

A function/method should do one thing. If it then needs to be applied to a dictionary or list kt can be.

To which end the `for dna_num in sorted(traces.keys())` should be moved out of this method and it can be made a `@staticmethod` that takes a single `np.ndarray` and line 93 in `trace_dna()` moved into the preceeding loop.

The same is true of all of the methods outside of that loop.

Reply:
My aim with this PR was only to refactor the dnatracing and skeletonisation methods necessary to continue with the knots and cats work which Alice is presenting soon. Hence why I've only added the docstring here and not modified/refactored any of the code.

I'd expect this to be done in a future PR where the traces are refactored as I just don't have the time right now. This can then be merged alongside the skeleton refactor (this) and the curvature refactor (the one you're currently working on) to refactor all of dnatracing bit by bit.